securedrop/journalist_app/api2/__init__.py

```diff
@@ -131,11 +93,27 @@ def metadata() -> Response:
     the ``Reply`` tables for the set of all item UUIDs.
     """
     try:
-        requested = MetadataRequest(**request.json)  # type: ignore
+        requested = BatchRequest(**request.json)  # type: ignore
     except (TypeError, ValueError) as exc:
         abort(422, f"malformed request; {exc}")
 
-    response = MetadataResponse()
+    response = BatchResponse()
+
+    for event in requested.events:
+        """
+        # Case 1: already processed: If event.snowflake_id is already cached in
+        # Redis, we've already processed it; just return its status.
+        status = EventStatus(event.snowflake_id, 200)  # FIXME
+        response.events.append(status)
+        """
```

Comment:
Assumption: `event.id` snowflakes come from the client already parameterized by (hash of) the session token, so a Redis-backed TTL-expired cache of IDs is all we need to enforce idempotence.  Any concerns about this approach?

Reply:
That seems right to me, notably our redis eviction policy is the default "noeviction", so keys should never be removed except for when the TTL expires (if redis fills up, then it'll reject writes entirely, but I don't think we need to worry about that scenario too much, it would be a pretty extreme self-DoS)